src/lib/containers/Login.js

```diff
@@ -124,7 +134,68 @@ class Login extends React.Component {
         }
     }
 
+    componentDidMount() {
+        let code = new URL(window.location.href)
+        // in test environment the searchParams isn't defined
+        if (code.searchParams && (code = code.searchParams.get("code"))) {
+            SynapseClient.oAuthSessionRequest(GOOGLE_OAUTH, code , `http://localhost:3000/?provider=${GOOGLE_OAUTH}`).then(
```

Comment:
The final redirect target should be a config parameter to this component (since it will vary depending on the app).